ompi/mca/pml/ob1/pml_ob1_recvreq.c

```diff
@@ -983,14 +983,10 @@ int mca_pml_ob1_recv_request_schedule_once( mca_pml_ob1_recv_request_t* recvreq,
         } while(!size);
         btl = bml_btl->btl;
 
-        /* NTH: This conditional used to check if there was a registration in
-         * recvreq->req_rdma[rdma_idx].btl_reg. If once existed it was due to
-         * the btl not needed registration (equivalent to btl->btl_register_mem
-         * != NULL. This new check is equivalent. Note: I feel this protocol
-         * needs work to better improve resource usage when running with a
-         * leave pinned protocol. */
-        if (btl->btl_register_mem && (btl->btl_rdma_pipeline_frag_size != 0) &&
```

Comment:
Does anyone know why there used to be a check about register mem here?  I honestly can't figure it out.  @bosilca / @hjelmn?

Reply:
I assumed that someone wanted to prevent pipelining if we have support for RMA. For the problem at hand, if we don't enforce this limit we will need to alter the internals of the btl_tcp_send to cope with very large buffers.